### ISSUE TITLE
udid variable in device.rb is nil on first launch

### ISSUE THREAD

**Issue by JoeSSS** [calabash/calabash-ios#1058]
Starting from run-loop 2.1.0 this error randomly occurs on first launch:

```
NoMethodError: undefined method `[]' for nil:NilClass
from /usr/local/lib/ruby/gems/2.1.0/gems/run_loop-2.1.0/lib/run_loop/device.rb:209:in `physical_device?'
```

This happens because `udid `in ` !udid[DEVICE_UDID_REGEX, 0].nil?`  on that moment is nil. Right after it coming through this error and test failing udid get a value and tests are starting to pass.

**Comment by jmoody** [comment]
Reproduced locally.  I am not sure what the cause is yet, but I have some ideas.  A fix will be available in run-loop 2.1.1.

**Comment by jmoody** [comment]
Gotcha!

It looks like there is some anomalous output from instruments that we are not handling properly.  Sometimes it is there and sometimes it is now.

Should be easy to fix.

```
 udid is nil
  
  /Users/moody/git/calabash/run-loop/lib/run_loop/device.rb:210:in `physical_device?'
  /Users/moody/git/calabash/run-loop/lib/run_loop/device.rb:189:in `instruments_identifier'
  /Users/moody/git/calabash/run-loop/lib/run_loop/device.rb:115:in `block in device_with_identifier'
  /Users/moody/git/calabash/run-loop/lib/run_loop/device.rb:114:in `each'
  /Users/moody/git/calabash/run-loop/lib/run_loop/device.rb:114:in `detect'
  /Users/moody/git/calabash/run-loop/lib/run_loop/device.rb:114:in `device_with_identifier'
  
     name: == Devices ==
  version: 2.9
   (RuntimeError)
  /Users/moody/git/calabash/run-loop/lib/run_loop/device.rb:211:in `physical_device?'
  /Users/moody/git/calabash/run-loop/lib/run_loop/device.rb:189:in `instruments_identifier'
  /Users/moody/git/calabash/run-loop/lib/run_loop/device.rb:115:in `block in device_with_identifier'
  /Users/moody/git/calabash/run-loop/lib/run_loop/device.rb:114:in `each'
  /Users/moody/git/calabash/run-loop/lib/run_loop/device.rb:114:in `detect'
  /Users/moody/git/calabash/run-loop/lib/run_loop/device.rb:114:in `device_with_identifier'
  /Users/moody/git/calabash/run-loop/lib/run_loop/device.rb:162:in `detect_device'
  /Users/moody/git/calabash/run-loop/lib/run_loop/core.rb:71:in `run_with_options'
  /Users/moody/git/calabash/run-loop/lib/run_loop.rb:130:in `run'
  /Users/moody/git/calabash/calabash-ios/calabash-cucumber/lib/calabash-cucumber/launcher.rb:326:in `block in new_run_loop'
  /Users/moody/git/calabash/calabash-ios/calabash-cucumber/lib/calabash-cucumber/launcher.rb:324:in `times'
  /Users/moody/git/calabash/calabash-ios/calabash-cucumber/lib/calabash-cucumber/launcher.rb:324:in `new_run_loop'
  /Users/moody/git/calabash/calabash-ios/calabash-cucumber/lib/calabash-cucumber/launcher.rb:302:in `relaunch'
  ./features/support/01_launch.rb:69:in `Before'
```

**Comment by JoeSSS** [comment]
Just add a comment. Looks like it is not coming with 2.1.0. I wasn't get this error because I had `system 'xcrun instruments -s devices'` on first launch. I deleted this string when 2.1.0 was released, and now could get the error on 2.0.9 as well.

**Comment by jmoody** [comment]
@JosSSS  Can you consistently get this to pop?  If so, can we do a tmate session?  I have tracked down the problem to output of `xcrun simctl list devices` and not to the output of `instruments`.

**Comment by JoeSSS** [comment]
Hey, I was able to reproduce it on one of our machines from time to time (not by demand).  What about instruments, I don't see any anomalous output in `instruments`, just saying that I don't see this error anymore  when running instruments before tests.

**Comment by jmoody** [comment]
The problem is caused by new output from `simctl` in Xcode 7.3.

My guess is that calling `instruments` resets the CoreSimulatorService but does not log any output.  We know that `instruments` interacts with CoreSimulatorService because `-s devices` outputs available simulators and devices as well.

The reason this error occurs once and then not again, is that warning is not printed on the second call - the CoreSimulatorService has already resolved itself so no warning is generated.

### Sample output

```
2016-04-11 16:17:48.474 simctl[98400:32684208] CoreSimulator is attempting to unload a stale CoreSimulatorService job.  Existing job (com.apple.CoreSimulator.CoreSimulatorService.117.15.1.lkhDXxRPp5yy) is from an older version and is being removed to prevent problems.
2016-04-11 16:17:48.732 simctl[98400:32684208] CoreSimulator is attempting to unload a stale CoreSimulatorService job.  Existing job (com.apple.CoreSimulator.CoreSimulatorService.117.15.1.dltcnwGXJZL4) is from an older version and is being removed to prevent problems.
2016-04-11 16:17:48.988 simctl[98400:32684208] CoreSimulator is attempting to unload a stale CoreSimulatorService job.  Existing job (com.apple.CoreSimulator.CoreSimulatorService.179.1.E8ttyeDeVgWK) is from an older version and is being removed to prevent problems.
2016-04-11 16:17:49.245 simctl[98400:32684208] CoreSimulator is attempting to unload a stale CoreSimulatorService job.  Existing job (com.apple.CoreSimulator.CoreSimulatorService.191.4.1.sUuqCSHdtoM8) is from an older version and is being removed to prevent problems.
2016-04-11 16:17:49.512 simctl[98400:32684208] CoreSimulator is attempting to unload a stale CoreSimulatorService job.  Detected Xcode.app relocation or CoreSimulatorService version change.  Framework path (/Xcode/7.3/Xcode.app/Contents/Developer/Library/PrivateFrameworks/CoreSimulator.framework) and version (209.19) does not match existing job path (/Xcode/7.2.1/Xcode.app/Contents/Developer/Library/PrivateFrameworks/CoreSimulator.framework/Versions/A/XPCServices/com.apple.CoreSimulator.CoreSimulatorService.xpc/Contents/MacOS/com.apple.CoreSimulator.CoreSimulatorService) and version (201.3).
2016-04-11 16:17:49.770 simctl[98400:32684208] Failed to locate a valid instance of CoreSimulatorService in the bootstrap.  Adding it now.
```

**Comment by jmoody** [comment]
Fixed in run-loop develop.

**Comment by jmoody** [comment]
@JoeSSS Should be fixed in the 0.19.0.pre1 release.  However, there are some problems with dismissing alerts - I will be pushing a pre2 release tomorrow.
